src/socket.rs

```diff
@@ -109,6 +114,38 @@ impl Socket {
         self.url.clone()
     }
 
+    pub fn is_connected(&self) -> bool {
+        self.status() == Status::Connected
+    }
+
+    pub fn is_waiting_to_reconnect(&self) -> bool {
+        self.status() == Status::WaitingToReconnect
+    }
+
+    pub fn has_never_connected(&self) -> bool {
+        self.status() == Status::NeverConnected
+    }
+
+    pub fn is_disconnected(&self) -> bool {
+        self.status() == Status::Disconnected
+    }
+
+    pub fn is_shutting_down(&self) -> bool {
+        self.status() == Status::ShuttingDown
+    }
+
+    pub fn is_shutdown(&self) -> bool {
+        self.status() == Status::ShutDown
+    }
+
+    pub fn status(&self) -> Status {
+        self.status.get()
```

Comment:
This would also work as Guard<_, State>. See https://docs.rs/arc-swap/latest/arc_swap/struct.ArcSwapAny.html#method.load